src/main/kotlin/brs/db/sql/SqlTransactionDb.kt

```diff
@@ -141,36 +141,62 @@ internal class SqlTransactionDb(private val dp: DependencyProvider) : Transactio
-                        version = transaction.version
-                        hasMessage = transaction.message != null
-                        hasEncryptedMessage = transaction.encryptedMessage != null
-                        hasPublicKeyAnnouncement = transaction.publicKeyAnnouncement != null
-                        hasEncrypttoselfMessage = transaction.encryptToSelfMessage != null
-                        ecBlockHeight = transaction.ecBlockHeight
-                        if (transaction.ecBlockId != 0L) ecBlockId = transaction.ecBlockId
-                        changed(true)
-                    }
-                }).execute()
+                var insertQuery = ctx.insertInto(
+                    TRANSACTION,
+                    TRANSACTION.ID,
+                    TRANSACTION.DEADLINE,
+                    TRANSACTION.SENDER_PUBLIC_KEY,
+                    TRANSACTION.RECIPIENT_ID,
+                    TRANSACTION.AMOUNT,
+                    TRANSACTION.FEE,
+                    TRANSACTION.REFERENCED_TRANSACTION_FULLHASH,
+                    TRANSACTION.HEIGHT,
+                    TRANSACTION.BLOCK_ID,
+                    TRANSACTION.SIGNATURE,
+                    TRANSACTION.TIMESTAMP,
+                    TRANSACTION.TYPE,
+                    TRANSACTION.SUBTYPE,
+                    TRANSACTION.SENDER_ID,
+                    TRANSACTION.ATTACHMENT_BYTES,
+                    TRANSACTION.BLOCK_TIMESTAMP,
+                    TRANSACTION.FULL_HASH,
+                    TRANSACTION.VERSION,
+                    TRANSACTION.HAS_MESSAGE,
+                    TRANSACTION.HAS_ENCRYPTED_MESSAGE,
+                    TRANSACTION.HAS_PUBLIC_KEY_ANNOUNCEMENT,
+                    TRANSACTION.HAS_ENCRYPTTOSELF_MESSAGE,
+                    TRANSACTION.EC_BLOCK_HEIGHT,
+                    TRANSACTION.EC_BLOCK_ID
+                )
+                transactions.map { transaction ->
+                    insertQuery = insertQuery.values(
+                        transaction.id,
+                        transaction.deadline,
+                        transaction.senderPublicKey,
+                        transaction.recipientId,
+                        transaction.amountPlanck,
+                        transaction.feePlanck,
+                        transaction.referencedTransactionFullHash,
+                        transaction.height,
+                        transaction.blockId,
+                        transaction.signature,
+                        transaction.timestamp,
+                        transaction.type.type,
+                        transaction.type.subtype,
+                        transaction.senderId,
+                        getAttachmentBytes(transaction),
+                        transaction.blockTimestamp,
+                        transaction.fullHash,
+                        transaction.version,
+                        transaction.message != null,
```

Comment:
!transaction.message ?

Reply:
this field is hasMessage so needs a null comparison. ! as an operator only works on bools so would not work here